### PR TITLE
8.0 small enhancements on mis_builder

### PR DESCRIPTION
**Pull request by alexis-via**
Add ability to use ('user_id', '=', uid) in a domain, which is important for dashboards for salesman

### CODE REVIEW

File: mis_builder/models/mis_builder.py [hunk 1]
```diff
@@ -447,7 +447,9 @@ def _fetch_queries(self):
         res = {}
         for query in self.report_instance_id.report_id.query_ids:
             model = self.env[query.model_id.model]
-            domain = query.domain and safe_eval(query.domain) or []
+            domain = query.domain and safe_eval(
+                query.domain,
+                {'uid': self._uid, 'context': self._context}) or []
```
Comment (lmignon) [inline mis_builder/models/mis_builder.py:452]: Why `self._uid` and not `self.env.uid` ?
Why do you want to pass the context?

**Comment by alexis-via** [comment]
@lmignon 

> Why self._uid and not self.env.uid ?
> I guess it's just the fact that I'm too lazy to type 3 additionnal characters... and, in the battle againt the 79 characters max, 3 additionnal characters is a significant saving of nearly 4% ! :)
> 
> Why do you want to pass the context?
> I don't know. I was just think "what could be usefull in additional to "uid" ? I thought some people might find a scenario where they want to access the context in here... I don't have such a scenario for the moment.

**Comment by lmignon** [comment]
@alexis-via IMO it's better to use self.env.uid and self.env.context to avoid to rely on an `implementation detail`.  :smirk:

**Comment by alexis-via** [comment]
@lmignon 
For me, self._uid and self._context is an official shortcut, cf the slides 16 of this official presentation of @rco-odoo
http://fr.slideshare.net/openobject/odoo-from-v7-to-v8-the-new-api
But, I you still prefer the "long" notation, I'll change it :)

**Comment by alexis-via** [comment]
One idea:  do you think we could "plug" in the code of odoo that "evals" domain for ir.actions.act_windows ? For example, for the domains in the XML code of ir.actions.act_windows, we can put domains such as:
('date', '>=', (context_today() - datetime.timedelta(days=365)).strftime('%Y-%m-%d'))
So it opens more possibilities...

**Comment by alexis-via** [comment]
In the master branch:
odoo-master % grep -ri "self._uid" *|wc -l
95

So it's widely used, even in the official addons

**Comment by alexis-via** [comment]
odoo-master % grep -ri "self._context" *|wc -l
189

**Comment by lmignon** [comment]
@alexis-via OK for self._...

**Comment by sbidoul** [comment]
@alexis-via I examined your idea to use the same context as in ir_action.
https://github.com/odoo/odoo/blob/8.0/openerp/addons/base/ir/ir_actions.py#L942

Would you update your PR to add in the evaluation context:
- time
- datetime
- dateutil
- env

Note the other are marked deprecated (uid, context, user, ...). Some are not relevant in this case (model, workflow, Warning).

**Pull request merged**